usersService/src/main/java/il/ac/afeka/usersservice/logic/ReactiveUsersService.java

```diff
@@ -65,16 +67,30 @@ public Flux<UserBoundary> getAll() {
     @Override
     public Mono<UserBoundary> GetUserByEmail(String email, String password) {
         return userCrud
-                .findByEmailAndPassword(email, password)
+                .findById(email)
+                .filter(user -> PasswordManager.verify(password, user.getPassword()))
```

Comment:
if password not right need to raise error. if im not mistaken i will return nothing due to filter.

Reply:
It will return nothing you are correct, but that's what Eyal was asking for.